From 3a384047123abc4a4fe2e81eb5dc4050746aa9e8 Mon Sep 17 00:00:00 2001
From: Tomas Bzatek <tbzatek@redhat.com>
Date: Fri, 17 Jan 2014 16:10:33 +0000
Subject: udisks_daemon_util_get_caller_uid_sync(): Add missing goto

Signed-off-by: David Zeuthen <zeuthen@gmail.com>
---
diff --git a/src/udisksdaemonutil.c b/src/udisksdaemonutil.c
index b3a3acb..a588580 100644
--- a/src/udisksdaemonutil.c
+++ b/src/udisksdaemonutil.c
@@ -840,6 +840,7 @@ udisks_daemon_util_get_caller_uid_sync (UDisksDaemon            *daemon,
                        UDISKS_ERROR,
                        UDISKS_ERROR_FAILED,
                        "User with uid %d does not exist", (gint) uid);
+          goto out;
         }
       else if (pw == NULL)
         {
--
cgit v0.9.0.2-2-gbebe
