From 951472a6f5914bbc56587d56e97fa40ff1908794 Mon Sep 17 00:00:00 2001
From: Tomas Bzatek <tbzatek@redhat.com>
Date: Wed, 22 Jan 2014 14:42:09 +0100
Subject: [PATCH 2/3] udiskslinuxmanager.c: Don't use uninitialized wait_data
 struct

---
 src/udiskslinuxmanager.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/udiskslinuxmanager.c b/src/udiskslinuxmanager.c
index d15eba6..53703d1 100644
--- a/src/udiskslinuxmanager.c
+++ b/src/udiskslinuxmanager.c
@@ -538,7 +538,6 @@ handle_mdraid_create (UDisksManager         *_object,
 {
   UDisksLinuxManager *manager = UDISKS_LINUX_MANAGER (_object);
   UDisksObject *array_object = NULL;
-  WaitForArrayData wait_data;
   uid_t caller_uid;
   GError *error = NULL;
   const gchar *message;
@@ -778,8 +777,8 @@ handle_mdraid_create (UDisksManager         *_object,
   if (array_object == NULL)
     {
       g_prefix_error (&error,
-                      "Error waiting for array object after creating /dev/md%d",
-                      wait_data.md_num);
+                      "Error waiting for array object after creating %s",
+                      raid_device_file);
       g_dbus_method_invocation_take_error (invocation, error);
       goto out;
     }
-- 
1.8.1.5

